docs: create ix-video hosted docs PE-2332

luqven opened the pull request:
This PR creates a VuePress application inside the project that generates a documentation static site.

TODO:
- [x] Document the attributes
- [x] Document auto-sizing
- [x] Document fixed sizing
- [x] Document styling overrides
- [x] change web-components references to ix-video
- [x] Add a deploy script -- in follow-up PR

## 📹 Video

https://user-images.githubusercontent.com/16711614/169896927-4611c485-2dac-48ee-9153-95d59b199d5e.mov





<!---GHSTACKOPEN-->
### Stacked PR Chain: PE-2332
| PR | Title | Status |  Merges Into  |
|:--:|:------|:-------|:-------------:|
|#1|docs: update readme and create documentation PE-2332|![](https://img.shields.io/github/pulls/detail/state/imgix/ix-video/1?label=Approved)|-|
|#2|👉 docs: create ix-video hosted docs PE-2332|![](https://img.shields.io/github/pulls/detail/state/imgix/ix-video/2?label=Approved)|#1|
|#3|docs: rename vuepress to docs PE-2332|![](https://img.shields.io/github/pulls/detail/state/imgix/ix-video/3?label=Approved)|#2|
|#4|docs(release): v1.0.0-rc.1 PE-2332|![](https://img.shields.io/github/pulls/detail/state/imgix/ix-video/4?label=Approved)|#3|
<!---GHSTACKCLOSE-->

luqven committed:
docs: create ix-video hosted docs

sherwinski commented on vuepress/package.json:
```diff
@@ -0,0 +1,19 @@
+  "repository": "@imgix/web-components/docs",
+  "scripts": {
+    "dev": "vuepress dev src",
+    "build": "vuepress build src"
```
Just FYI, these commands don't run locally unless you have `vuepress` installed

sherwinski commented on vuepress/src/index.md:
```diff
@@ -0,0 +1,15 @@
+---
+home: true
+heroImage: https://pluginicons.craft-cdn.com/imgix.svg?1511197213
```
out of curiosity, where did you get this image?

luqven replied:
I honestly don't recall now... maybe the contenful plugins page? I should probably replace this 😬

sherwinski commented on vuepress/src/index.md:
```diff
@@ -0,0 +1,15 @@
+  - title: Bite sized
+    details: Small bundle with few dependencies
+  - title: Simple
+    details: Best practices use out-of-the-box
```
nit: maybe the wording could be a bit clearer
```suggestion
    details: Uses best practices out-of-the-box
```
Or something to that effect

sherwinski commented on vuepress/src/installation/usage.md:
```diff
@@ -0,0 +1,102 @@
+# Usage
+
+This library exports a collection of Web Components. Bellow is an example of how
```
```suggestion
This library exports a collection of Web Components. Below is an example of how
```

luqven replied:
One of these days I will learn how to spell "below" 🤦🏻‍♂️

sherwinski commented on vuepress/src/installation/usage.md:
````diff
@@ -0,0 +1,102 @@
+</template>
+```
+
+Vue might show some warnings in the console if you don't declare your custom
+element in it's configuration. To turn these warnings off, add the following:
````
```suggestion
You will need to signal to the Vue compiler that this component is a custom element by adding the following:

```js
//vue.config.js
module.exports = {
  chainWebpack: config => {
    config.module
      .rule('vue')
      .use('vue-loader')
      .tap(options => ({
        ...options,
        compilerOptions: {
          // treat any tag that starts with ix- as custom elements
          isCustomElement: (name) => name.startsWith('ix-'),
        }
      })
    )
  }
```
```

sherwinski commented on vuepress/src/installation/usage.md:
````diff
@@ -0,0 +1,102 @@
+Vue might show some warnings in the console if you don't declare your custom
+element in it's configuration. To turn these warnings off, add the following:
+
+```js
````
```suggestion
Or if you're using Vite, you can add the following to your `vite.config.js`:

```js
```

sherwinski commented on vuepress/src/overview/development.md:
```diff
@@ -0,0 +1,91 @@
+
+## Bundling and minification
+
+This project uses Rollup to bundle and minify the TypeScript component files
+into a single file in `dist/`. The rollup config is located at
+`rollup.config.js`.
```
Not a big deal but just wanted to point out the unusual spacing in these three lines. But it might be for a specific reason that I can't see 🤷 
```suggestion
This project uses Rollup to bundle and minify the TypeScript component files into a single file in `dist/`. The rollup config is located at `rollup.config.js`.
```

luqven replied:
Huh... that's strange. Not sure what happenned there good catch 👍🏻

sherwinski commented on vuepress/src/overview/ix-video.md:
```diff
@@ -0,0 +1,293 @@
+
+Vue has great support for Web Components, so you can use the component in your Vue application without any additional steps.
+
+However, Vue might show some warnings in the console if you don't declare your custom element in it's configuration. To turn these warnings off, you can add the following:
```
```suggestion
You will need to signal to the Vue compiler that this component is a custom element by adding the following:

```js
//vue.config.js
module.exports = {
  chainWebpack: config => {
    config.module
      .rule('vue')
      .use('vue-loader')
      .tap(options => ({
        ...options,
        compilerOptions: {
          // treat any tag that starts with ix- as custom elements
          isCustomElement: (name) => name.startsWith('ix-'),
        }
      })
    )
  }
  
  Or if you're using Vite, you can add the following to your `vite.config.js`:
```

sherwinski replied:
On that note, do we still need to include these examples in these files as well? We already include them on the usage page.

luqven replied:
True, I think they're safe to remove.

luqven replied:
The usage page wasn't being linked/rendered anywhere 🤦🏼 . Moved its content into the main readme for now.

luqven committed:
docs: address feedback

luqven committed:
docs(vuepress): move usage section into ix-video md

The `usage.md` wasn't acutally linked/rendered by anything. Now the
section is rendered inside the ix-video document.

luqven committed:
chore: change wc references to ix-video

luqven committed:
chore: update installation for static usage

luqven committed:
chore: update hero image

luqven committed:
build: add docs dev script

luqven committed:
docs: move ix-video vue docs into sep md files

luqven committed:
docs: add vue installation docs reqs section

luqven committed:
docs: update vue readme to reflects new routes

luqven committed:
docs: update container example to highlight fixed

luqven committed:
docs: update vue routes

luqven committed:
docs: change cust h to h1 from h2

luqven committed:
docs: move md files into vuepress docs

luqven committed:
docs(readme): change docs links to vuepress directory

luqven committed:
docs: rename vuepress to docs

luqven committed:
docs: move bundle import to fix ssr issues

luqven committed:
build: add docs deploy script

luqven committed:
docs: fix broken quick-start links

luqven committed:
build: change docs scripts dir ref

luqven committed:
docs(release): v1.0.0-rc.1

luqven committed:
docs: update readme docs links

sherwinski reviewed (changes requested):
I confirmed that `npm run docs:dev` only works if you cd into the vuepress directory first and run `npm install`. Not sure what's the best solution there but it feels like the experience could be a bit better; maybe consider adding a `docs:dev:setup` step as well?
Edit: ah maybe this did get addressed in https://github.com/imgix/ix-video/pull/3 ?

Also, I remembered that we discussed offline adding the same warning to the `width` attribute section as the in `height` section. I didn't see it in this PR but thought I'd point it out in case it slipped through the cracks.

luqven commented:
> Edit: ah maybe this did get addressed in https://github.com/imgix/ix-video/pull/3 ?

Not really, unless this is blocking I'm going to tackle this in follow-up PR. There might be a way to use `npx vuepress` here.

> we discussed offline adding the same warning...

💯 , let me triple-check that isn't in the follow-up PR. If not it's def missing.

luqven committed:
docs: add warning and update build scripts

luqven commented:
@sherwinski addressed your feedback in #3 . Going to go ahead and merge these in so I can cut a release later today.

sherwinski reviewed (approved):
With the updates made in #3, everything LGTM now 👍
